### PR TITLE
[Container Section 1 of 2] DCOS-12398: Align container section with doc specification

### PR DESCRIPTION
**Pull request by mlunoe**
This PR aligns the two container sections according to the docs specification.

Container Section 1 of 2: https://github.com/dcos/dcos-ui/pull/1770
Container Section 2 of 2: https://github.com/dcos/dcos-ui/pull/1771

### CODE REVIEW

**Comment by mlunoe** [comment]
Rebased with integration test fixes

**Comment by mlunoe** [comment]
Restest this please

**Comment by mlunoe** [comment]
Restest this please

**Comment by mlunoe** [comment]
Run integration tests

**Comment by mlunoe** [comment]
Run integration tests

**Comment by wavesoft** [comment]
Hey @mlunoe , click ing on "+ ADD ARTIFACT" on a container is not doing anything, without throwing an error!

**Comment by mlunoe** [comment]
Fixed!

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 1]
```diff
@@ -23,92 +23,100 @@ const {DOCKER, NONE, MESOS} = ContainerConstants.type;
...
+        <FormRow key={`${path}.artifacts.${index}`}>
           <FormGroup
             className="column-10"
-            showError={Boolean(errors[index])}>
+            showError={Boolean(artifactErrors[index])}>
             {label}
             <FieldInput
               name={`fetch.${index}.uri`}
-              type="text"
+              placeholder="http://"
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: This placeholder is confusing. Is it telling the user that their URI must start with `http://`? Or is it telling me that I can exclude the `http://`? Either way let's add some instructional text here.

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 2]
```diff
@@ -23,92 +23,100 @@ const {DOCKER, NONE, MESOS} = ContainerConstants.type;
...
               value={item.uri}/>
-            <FieldError>{errors[index]}</FieldError>
+            <FieldError>{artifactErrors[index]}</FieldError>
           </FormGroup>
-          <FormGroup className="flex flex-item-align-end column-2 flush-left">
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Are we sure we want to remove the `flush-left` class here? We're inconsistent with this (which means we need to find a better solution for this, in a separate PR), but it looks the majority of `DeleteRowButtons` wrappers have `flush-left` in addition to their column classes.

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 3]
```diff
@@ -241,107 +266,82 @@ class ContainerServiceFormSection extends Component {
...
-    const containerType = findNestedPropertyInObject(
-      data,
-      'container.type'
+      `${path}.docker.image`
     );
 
     return (
       <div>
-        <h2 className="short-bottom">
+        <h2 className="short-top short-bottom">
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: These classes should be `flush-top short-bottom`

Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: I just realized this is used in both the multi-container form and single container form... In the single-container form, the classes should be just `short-bottom` and in the multi-container form it should be `flush-top short-bottom`... not sure how to accomplish this cleanly.

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 1 above)
Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: This is the instructions from the docs team. @sascala What is your recommendation here?

Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Ahh okay, I'm happy for my opinion to be superseded by the docs team :)

Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: How about `http://example.com`?

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 2 above)
Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Done!

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 1 above)
Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: I think the full URL example is better than `http://`, personally. But this change doesn't have to block the merge of this PR — let's use the docs team as the source of truth here and update this if they think it's necessary.

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 3 above)
Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Let me handle that difference in the follow-up PR (where we merge the two sections), then

Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Handled here: https://github.com/dcos/dcos-ui/pull/1771/files#diff-1407574c69695d08ed04a5d46d0314fcR419

Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Nice! Thanks for the link 👍

**Comment by jfurrow** [comment]
Let's do it! 🚢

**Comment by orlandohohmeier** [comment]
invariant-violation
I get an error when "removing" an artifact from the JSON...

**Comment by mlunoe** [comment]
Nice find! I was not accessing errors correctly in the Pod section for artifacts!
This is fixed now:

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 4]
```diff
@@ -23,92 +23,100 @@ const {DOCKER, NONE, MESOS} = ContainerConstants.type;
...
-
 class ContainerServiceFormSection extends Component {
-  getArtifactsInputs(data = []) {
-    const errors = this.props.errors.fetch || [];
+  getArtifactsLabel() {
+    const tooltipContent = (
+      <span>
+        {'If your service requires additional files and/or archives of files, enter their URIs to download and, if necessary, extract these resources. '}
+        <a
+          href="https://mesosphere.github.io/marathon/docs/application-basics.html"
```
Comment (MatApple) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: This link doesn't tell me anything about artifacts. Do we have this in the marathon docs yet?

Comment (sascala) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: I think this is a better link: https://mesosphere.github.io/marathon/docs/artifact-store.html

Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Arh! Good catch! The docs were wrong

Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Fixed

**Comment by orlandohohmeier** [comment]
run integration tests

**Comment by wavesoft** [comment]
run unit tests

**Comment by wavesoft** [comment]
run unit tests

**Comment by wavesoft** [comment]
Run unit tests

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ class ContainerServiceFormSection extends Component {
       <span>
         {'If your service requires additional files and/or archives of files, enter their URIs to download and, if necessary, extract these resources. '}
         <a
-          href="https://mesosphere.github.io/marathon/docs/application-basics.html"
+          href="https://mesosphere.github.io/marathon/docs/artifact-store.html"
```
Comment (orlandohohmeier) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: This is not the correct link. It's about the Artifact Store and doesn't detail how to configure `fetch` (aka. `uris`). Please use `https://mesosphere.github.io/marathon/docs/application-basics.html` which at least touches the matter. 

@sascala do we have a better doc somewhere than the general app config doc?

Comment (mlunoe) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Ok, reverted back to what it was before

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 4 above)
Comment (orlandohohmeier) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: Please see: https://github.com/dcos/dcos-ui/pull/1770#discussion_r97770811

**Comment by orlandohohmeier** [comment]
Great job! ✨ 💯  Feel free to merge this one once the test pass...

File: plugins/services/src/js/components/forms/ContainerServiceFormSection.js (hunk 5 above)
Comment (sascala) [inline plugins/services/src/js/components/forms/ContainerServiceFormSection.js]: That makes sense -- pardon the mistake.

**Comment by MatApple** [comment]
run integration tests

**Pull request merged**